libs/community/langchain_community/llms/self_hosted.py

```diff
@@ -12,57 +11,66 @@
 logger = logging.getLogger(__name__)
 
 
-def _generate_text(
-    pipeline: Any,
-    prompt: str,
-    *args: Any,
-    stop: Optional[List[str]] = None,
-    **kwargs: Any,
-) -> str:
-    """Inference function to send to the remote hardware.
+class ModelPipeline:
+    def __init__(self, model_id, **model_kwargs):
+        super().__init__()
+        self.model_id, self.model_kwargs = model_id, model_kwargs
+        self.tokenizer, self.model, self.curr_pipeline, self.task = None, None, None, model_kwargs.get("task", None)
 
-    Accepts a pipeline callable (or, more likely,
-    a key pointing to the model on the cluster's object store)
-    and returns text predictions for each document
-    in the batch.
-    """
-    text = pipeline(prompt, *args, **kwargs)
-    if stop is not None:
-        text = enforce_stop_tokens(text, stop)
-    return text
-
-
-def _send_pipeline_to_device(pipeline: Any, device: int) -> Any:
-    """Send a pipeline to a device on the cluster."""
-    if isinstance(pipeline, str):
-        with open(pipeline, "rb") as f:
-            pipeline = pickle.load(f)
-
-    if importlib.util.find_spec("torch") is not None:
-        import torch
-
-        cuda_device_count = torch.cuda.device_count()
-        if device < -1 or (device >= cuda_device_count):
-            raise ValueError(
-                f"Got device=={device}, "
-                f"device is required to be within [-1, {cuda_device_count})"
-            )
-        if device < 0 and cuda_device_count > 0:
-            logger.warning(
-                "Device has %d GPUs available. "
-                "Provide device={deviceId} to `from_model_id` to use available"
-                "GPUs for execution. deviceId is -1 for CPU and "
-                "can be a positive integer associated with CUDA device id.",
-                cuda_device_count,
+    def load_model(self, hf_token) -> Any:
+        """
+        Accepts a huggingface model_id and returns a pipeline for the task.
+        Sent to the remote hardware and being executed there, as part of the rh.Module(LangchainLLMModelPipeline).
+        """
+        from transformers import AutoTokenizer, AutoModel, AutoModelForMaskedLM
+        from transformers import pipeline as hf_pipeline
+
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_id,
+                                                       token=hf_token)
+
+        try:
+            self.model = AutoModel.from_pretrained(self.model_id,
+                                                   token=hf_token)
+        except RuntimeError:
+            self.model = AutoModelForMaskedLM.from_pretrained(self.model_id,
```

Comment:
Not sure i totally follow - if we can't load as a generic `AutoModel` we will always be able to with `AutoModelForMaskedLM`?